nautobot_golden_config/utilities/helper.py

```diff
@@ -139,3 +150,157 @@ def get_device_to_settings_map(queryset):
+    return device_data
+
+
+class RenderConfigToPushError(Exception):
+    """Exception related to Render Configuration to Push operations."""
+
+
+def render_secrets(config_to_push: str, configs: models.GoldenConfig, request: HttpRequest) -> str:
+    """Renders secrets using the get_secrets filter.
+
+    This method is defined to render an already rendered intended configuration, but which have used the Jinja
+    `{% raw %}` tag to skip the first render (because the first one gets persisted, and for secrets we don't want it).
+    It also support chaining with some Netutils encrypt filters.
+
+    .. rubric:: Example Jinja render_secrets filters usage
+    .. highlight:: jinja
+    .. code-block:: jinja
+        ppp pap sent-username {{ secrets_group["slug"] | get_secret_by_secret_group_slug("password") | encrypt_type7 }}
+
+    Returns:
+        str : Return a string, with the rendered intended configuration with secrets, or an error message.
+
+    """
+    if not config_to_push:
+        return ""
+
+    jinja_env = jinja2.Environment(autoescape=True)
+
+    for name, func in jinja2_convenience_function().items():
+        # Only importing the encrypt helpers as complements to get_secrets filter
+        if name in ["encrypt_type5", "encrypt_type7"]:
+            jinja_env.filters[name] = func
+
+    # Wrapper for get_secret filter that includes user argument to ensure
+    # that secrets are only rendered by authorized users.
+    # To call this method, the view verifies that it's an authenticated request.
+    jinja_env.filters["get_secret_by_secret_group_slug"] = partial(get_secret_by_secret_group_slug, request.user)
+
+    try:
+        template = jinja_env.from_string(config_to_push)
+    except jinja_errors.TemplateAssertionError as error:
+        return f"Jinja encountered an TemplateAssertionError: '{error}'; check the template for correctness"
+
+    device_data = _get_device_agg_data(configs.device, request)
+
+    try:
+        return template.render(device_data)
+
+    except jinja_errors.UndefinedError as error:
+        raise RenderConfigToPushError(
+            f"Jinja encountered and UndefinedError: {error}, check the template for missing variable definitions.\n"
+        ) from error
+    except jinja_errors.TemplateSyntaxError as error:  # Also catches subclass of TemplateAssertionError
+        raise RenderConfigToPushError(
+            f"Jinja encountered a SyntaxError at line number {error.lineno},"
+            f"check the template for invalid Jinja syntax.\n"
+        ) from error
+    except jinja_errors.TemplateError as error:  # Catches all remaining Jinja errors
+        raise RenderConfigToPushError(
+            "Jinja encountered an unexpected TemplateError; check the template for correctness\n"
+        ) from error
+
+
+def get_config_to_push(
+    configs: models.GoldenConfig, request: HttpRequest, custom_subscribed: Optional[List[str]] = None
```

Comment:
Can remove custom_subscribed from signature, if in the future we wanted to have user provided, we would do it within the request.